modules/thermal_hydraulics/src/base/Simulation.C

```diff
@@ -633,7 +633,7 @@ Simulation::sortAddedComponentVariables() const
 void
 Simulation::addVariables()
 {
-  Transient * trex = dynamic_cast<Transient *>(getApp().getExecutioner());
+  TransientBase * trex = dynamic_cast<TransientBase *>(getApp().getExecutioner());
```

Comment:
Maybe same here for the dynamic_cast. Feel free to disregard

Reply:
The cast is used for differentiating between steady and transient behavior later on this.